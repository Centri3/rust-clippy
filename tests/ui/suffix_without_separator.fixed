//@run-rustfix
//@aux-build:proc_macros.rs:proc-macro
#![allow(unused)]
#![warn(clippy::suffix_without_separator)]

#[macro_use]
extern crate proc_macros;

fn main() {
    _ = 1u32;
    _ = 1u8;
    _ = (1u8);
    _ = [1u8, 2u8];
    _ = 1.0f64;
    _ = 1f64;
    // Do not lint
    _ = 1u32;
    _ = 1u8;
    _ = (1u8);
    _ = [1u8, 2u8];
    _ = 1.0f64;
    _ = 1f64;
    external! {
        _ = 1_u32;
        _ = 1_u8;
        _ = (1_u8);
        _ = [1_u8, 2_u8];
        _ = 1.0_f64;
        _ = 1_f64;
    }
    with_span! {
        span
        _ = 1_u32;
        _ = 1_u8;
        _ = (1_u8);
        _ = [1_u8, 2_u8];
        _ = 1.0_f64;
        _ = 1_f64;
    }
}
