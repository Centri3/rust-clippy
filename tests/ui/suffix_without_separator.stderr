error: this literal has a separator before its suffix
  --> $DIR/suffix_without_separator.rs:10:10
   |
LL |     _ = 1_u32;
   |          ^ help: remove it
   |
   = note: `-D clippy::suffix-without-separator` implied by `-D warnings`

error: this literal has a separator before its suffix
  --> $DIR/suffix_without_separator.rs:11:10
   |
LL |     _ = 1_u8;
   |          ^ help: remove it

error: this literal has a separator before its suffix
  --> $DIR/suffix_without_separator.rs:12:11
   |
LL |     _ = (1_u8);
   |           ^ help: remove it

error: this literal has a separator before its suffix
  --> $DIR/suffix_without_separator.rs:13:11
   |
LL |     _ = [1_u8, 2_u8];
   |           ^ help: remove it

error: this literal has a separator before its suffix
  --> $DIR/suffix_without_separator.rs:13:17
   |
LL |     _ = [1_u8, 2_u8];
   |                 ^ help: remove it

error: this literal has a separator before its suffix
  --> $DIR/suffix_without_separator.rs:14:12
   |
LL |     _ = 1.0_f64;
   |            ^ help: remove it

error: this literal has a separator before its suffix
  --> $DIR/suffix_without_separator.rs:15:10
   |
LL |     _ = 1_f64;
   |          ^ help: remove it

error: aborting due to 7 previous errors

